.pre-commit-config.yaml

```diff
@@ -0,0 +1,14 @@
+repos:
+-   repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v3.2.0
+    hooks:
+    -   id: trailing-whitespace
```

Comment:
Cool idea, what about adding more hooks?:

- check-merge-conflict 
- check-executables-have-shebangs
- check-shebang-scripts-are-executable
- detect-private-key
- mixed-line-ending

And a new hook repo for golangci-lint?

Reply:
ACK to the additions, I'm split between using the native call to `golangci-lint` from `pre-commit` verses the correct implementation where I'm just calling our `Makefile` `lint` target as eventually that could do more things we'd want coverage for here.

Reply:
If possible let's use our make target.

Reply:
ACK, it already is.